src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/jpa/Kpi.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.jpa
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.ReadOnlyProperty
+import org.springframework.data.annotation.Transient
+import org.springframework.data.domain.Persistable
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+data class Kpi(
+  @Id
+  val day: LocalDate,
+
+  val overdueReviews: Int,
+  val previousMonthReviewsConducted: Int,
+  val previousMonthPrisonersReviewed: Int,
+
+  @ReadOnlyProperty
+  val whenCreated: LocalDateTime? = null,
```

Comment:
from recollection, our "reactive" setup means you need to ` = LocalDateTime.now()` otherwise whenCreated and whenUpdated are not set…? or did it work for you?

Reply:
I've copied the model from the others. So it should be the setup that we made it work.

Reply:
the `IncentiveLevel`, for instance, has ` = LocalDateTime.now()`.

however `NextReviewDate` has ` = null` instead so at instantiation we have to do `whenUpdated = LocalDateTime.now(clock)`

Reply:
Just to recap our offline chat. Both fields are automatically set to `now()` correctly because of the DB table `DEFAULT now()` (similar to what happen for the next_review_date table/JPA entity).
If we were to update any of these records we'll have explicitly pass the `whenUpdated` field.

I don't think there is anything to do with this tbh.